lib/plugins/aws/lib/naming.js

```diff
@@ -381,6 +382,17 @@ module.exports = {
   getAlbTargetGroupLogicalId(functionName) {
     return `${this.getNormalizedFunctionName(functionName)}AlbTargetGroup`;
   },
+  getAlbTargetGroupNameTagValue(functionName) {
+    return `${
+      this.provider.serverless.service.service
+    }-${functionName}-${this.provider.getStage()}`;
+  },
+  getAlbTargetGroupName(functionName) {
+    return crypto
+      .createHash('md5')
+      .update(this.getAlbTargetGroupNameTagValue(functionName))
+      .digest('hex');
```

Comment:
Why hashing is needed? Wouldn't adding service name be good enough?

Reply:
Good point. I think it's necessary because of the max length for the name.

Reply:
And do we know what's allowed max length? Is it e.g. different from function name on AWS side?
In unhashed form iirc this name equals final function name (which is also combination of _service_, _base function name_ and _stage_ tokens)

Reply:
Max length is 32 (see PR description) and name has to be unique per service/function/stage, I think it would be hard to have a non-hash name that fits, even with truncating.

Reply:
Indeed, not sure why I didn't read through _Important_ note in description.

LGTM